lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -440,15 +440,7 @@ public Optional<TaskStatusProto> GetTaskStatus()
                 }
 
                 var taskStatusProto = _task.Value.GetStatusProto();
-                if (taskStatusProto.state == State.RUNNING)
-                {
-                    // only RUNNING status is allowed to rurn here, all other state pushed out to heartbeat 
-                    return Optional<TaskStatusProto>.Of(taskStatusProto);
-                }
-
-                var e = new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, "Task state must be RUNNING, but instead is in {0} state", taskStatusProto.state));
```

Comment:
We have observed in this part of the code comments which do not correspond to what the code actually does (for example, the `Send heartbeat such that user receives a TaskRunning message.` which started this whole fix). I wouldn't read too much into the comments here, because they can describe incorrect behavior of code. I don't think a method called `GetTaskStatus` should throw exception if the status is anything but running.